pulp_rpm/app/serializers.py

```diff
@@ -250,7 +251,8 @@ def deferred_validate(self, data):
         except OSError:
             raise NotAcceptable(detail='RPM file cannot be parsed for metadata.')
 
-        package = Package.objects.filter(**new_pkg)
+        attrs = {key: new_pkg[key] for key in chain.from_iterable(Package._meta.unique_together)}
```

Comment:
pulpcore provides a more structured way to get these fields (if I'm understanding your code right). Package should have a methods like these: https://github.com/pulp/pulpcore/blob/9aac516c851b6fce68dcc620b2d4fba5a4679d19/pulpcore/app/models/content.py#L264